dadab/dadab.js

```diff
@@ -3,77 +3,101 @@ based on:
-  var canvas = createCanvas(img.width, img.height);
-  canvas.parent("p5canvas");
+function setup () {
+  var canvas = createCanvas(img.width, img.height); // eslint-disable-line no-undef
+  canvas.parent('p5canvas');
 
   // allows the viewer to recreateCanvas and update surface to image file dimensions
   // surface.setResizable(true);
   // surface.setSize(img.width, img.height);
-  
+
+  // todo: tweak these settings
+  capture = createCapture(VIDEO);
+  capture.size(320, 240);
+  capture.hide();
+  setupGif();
+
   // load image onto surface
-  image(img, 0, 0);
+  // todo: fix the fact that `capture` hides the image
+  image(capture, 0, 0); // eslint-disable-line no-undef
 
   img.loadPixels();
-  noStroke();
- 
-   //set the speed at which rectangles are created; 1 frame = 1 rectangle
-  frameRate(30000);
+  noStroke(); // eslint-disable-line no-undef
+
+  // set the speed at which rectangles are created; 1 frame = 1 rectangle
+  frameRate(30000); // eslint-disable-line no-undef
 }
 
-function draw() {
-  // ellipse(100, 100, 100,100);
-  //
-  var randx = random(0,width);
-  var randy = random(0, height);
-  var randc = img.get(randx, randy);
-  
-  var randx_offset = random(0,8);
-  var randy_offset = random(0,8);
-  
-  /*var randh = random(5,7) + random(3,7);
-  var randw = randh * random(4,8);
-  var rands = random(2,5);*/
-  
-  //
-  var randh = random(2,3) + random(2,6);
-  var randw = randh * random(2,7);
-  var rands = random(2,6);
-  
-  //
-  fill(randc);
-  rect(randx+randx_offset, randy+randy_offset, randw, rands);
-  
-  //apply a posterized filter effect to the colors being randomly generated
+function draw () {
+  var randomX = random(0, width);
+  var randomY = random(0, height);
+  var randomColor = img.get(randomX, randomY);
+
+  var randomXOffset = random(0, 8);
+  var randomYOffset = random(0, 8);
+
+  var randomHeight = random(2, 3) + random(2, 6);
+  var randomWidth = randomHeight * random(2, 7);
+  var randomS = random(2, 6);
+
+  fill(randomColor);
+  rect(randomX + randomXOffset, randomY + randomYOffset, randomWidth, randomS);
+
+  // apply a posterized filter effect to the colors being randomly generated
   filter(POSTERIZE, 6);
-  /*filter(POSTERIZE, random(19,21));
-  filter(INVERT);*/
-  
-  console.log(frameCount);
-
-  // save frame 1500 - 3500 to a local folder 
-  // if (frameCount > 1500 && frameCount < 1800) {
-  //   saveFrames("/Users/dezein/Desktop/dadab-images/f#####.gif");
-  // }
-
-  //after 3500 frames stop the sketch from running
-  // if (frameCount == 3500) {
-  //   exit();
-  // }
+  /*
+  filter(POSTERIZE, random(19,21));
+  filter(INVERT);
+  */
+
+  if (frameCount > START_FRAME && frameCount < END_FRAME) {
+    console.log(`saving frame #${frameCount}`);
+    gif.addFrame(document.getElementById('defaultCanvas0'), {delay: 1, copy: true});
+  }
+
+  if (frameCount === END_FRAME) {
+    console.log('done');
+    // remove();
```

Comment:
Instead of removing the image, we could continue to show the glitch growing as the gif is getting saved, as it takes a little while to save it just locally -- this isn't even including the time it'll take to upload to Firebase and Twitter.